src/django_components/logger.py

```diff
@@ -1,3 +1,35 @@
 import logging
+from typing import Literal, Optional
 
 logger = logging.getLogger("django_components")
+
+
+def trace_msg(
+    action: Literal["PARSE", "ASSOC", "RENDR", "GET", "SET"],
+    node_type: Literal["COMP", "FILL", "SLOT", "IFSB", "N/A"],
+    node_name: str,
+    node_id: str,
+    msg: str = "",
+    component_id: Optional[str] = None,
+) -> None:
+    """
+    Log a tracing statement to `logger.debug` like so:
+
+    `"ASSOC SLOT test_slot ID 0088 TO COMP 0087"`
+    """
+    msg_prefix = ""
+    if action == "ASSOC":
+        if not component_id:
+            raise ValueError("component_id must be set for the ASSOC action")
+        msg_prefix = f"TO COMP {component_id}"
+    elif action == "RENDR" and node_type != "COMP":
+        if not component_id:
+            raise ValueError("component_id must be set for the RENDER action")
+        msg_prefix = f"FOR COMP {component_id}"
+
+    msg_parts = [f"{action} {node_type} {node_name} ID {node_id}", *([msg_prefix] if msg_prefix else []), msg]
+    full_msg = " ".join(msg_parts)
+
+    # NOTE: When debugging tests during development, it may be easier to change
+    # this to `print()`
+    logger.debug(full_msg)
```

Comment:
Not sure about this. I can see how debuging paths could be useful for a developer, but this is for us right? Since this will likely output a lot of text, it might make the other debug statement less useful. Should be change the other ones to info instead? Info = For developers, Debug = For us?

Reply:
Yeah, this is meant for us. I wouldn't want to bump up current debug to info, since debug level currently includes the info about which template names were matched against which files, and that too can be fairly verbose if there's a lot of files.

The `logger.debug` would've ideally been a TRACE level, but didn't search before if it's possible.

But I did now!

Supposedly something like this should do the trick:
```py
import logging

TRACE_LEVEL_NUM = 5  # Setting it lower than DEBUG which is 10

def trace(self, message, *args, **kws):
    if self.isEnabledFor(TRACE_LEVEL_NUM):
        self._log(TRACE_LEVEL_NUM, message, args, **kws)

logging.Logger.trace = trace
logging.addLevelName(TRACE_LEVEL_NUM, "TRACE")
```

So what we could have is:
- `trace` (LOG LEVEL 5) - This would be for us. This is where we would output the `trace_msg`.
- `debug` (LOG LEVEL 10) - This is for low-level or verbose output for users. This would contain info about template name resolution, as it does now.
- `info` (LOG LEVEL 20) - Regular messages for users. We don't use it currently.
-  We don't currently use others like `warn`, `error`, `critical` 

Lastly, in the snippet above, the bit where it does
```py
logging.Logger.trace = trace
logging.addLevelName(TRACE_LEVEL_NUM, "TRACE")
```

I assume we should probably first check if the "TRACE" level was already defined, and define it only it has not been yet.

Reply:
Sounds like a good plan to me! :)

Reply:
Implemented in https://github.com/EmilStenstrom/django-components/pull/437/commits/2c451693abd886cee180828513fead5726c51ad5 🎉 

Also in the end I omitted the following line as it felt a bit too invasive.
```py
logging.Logger.trace = trace
```